contracts/modules/SSVStaking.sol

```diff
@@ -0,0 +1,389 @@
+
+        StorageStaking storage s = SSVStorageStaking.load();
+
+        // 2. Update global and user states before balance change
+        _syncFees(s);
+        _settle(msg.sender, s);
+
+        // 3. Transfer SSV from user to this contract
+        if (!SSVStorage.load().token.transferFrom(msg.sender, address(this), amount)) {
+            revert TokenTransferFailed();
+        }
+
+        // 4. Update delegations (before minting cSSV to reflect the new weight)
+        _createDelegation(msg.sender, amount, s);
+
+        // 5. Mint cSSV receipt tokens 1:1
+        ICSSVToken(s.cssv).mint(msg.sender, amount);
+
+        emit Staked(msg.sender, amount);
+    }
+
+    function requestUnstake(uint256 amount) external {
+        if (amount == 0) revert ZeroAmount();
+
+        StorageStaking storage s = SSVStorageStaking.load();
+        address cssv = s.cssv;
+
+        // Ensure user doesn't have an existing pending request
+        if (s.withdrawals[msg.sender].amount != 0) revert CooldownActive();
+
+        // 1. Sync global state
+        _syncFees(s);
+        // 2. Settle user rewards using current balance (before burn)
+        uint256 bal = ICSSVToken(cssv).balanceOf(msg.sender);
+        _settleWithBalance(msg.sender, bal, s);
+        if (amount > bal) revert UnstakeAmountExceedsBalance();
+
+        // 3. Update delegations (remove weight proportional to amount)
+        _removeDelegation(msg.sender, amount, bal, s);
+
+        // 4. Burn cSSV tokens immediately
+        ICSSVToken(cssv).burn(msg.sender, amount);
+
+        // 5. Record pending withdrawal and set cooldown
+        uint64 unlockTime = uint64(block.timestamp + s.cooldownDuration);
+        s.withdrawals[msg.sender] = UnstakeRequest({amount: uint192(amount), unlockTime: unlockTime});
+
+        emit UnstakeRequested(msg.sender, amount, unlockTime);
+    }
+
+    function withdrawUnlocked() external {
+        StorageStaking storage s = SSVStorageStaking.load();
+        UnstakeRequest memory request = s.withdrawals[msg.sender];
+        uint256 amount = request.amount;
+        if (amount == 0) revert NothingToWithdraw();
+
+        // Verify cooldown period has passed
+        if (block.timestamp < request.unlockTime) revert CooldownNotFinished();
+
+        // Clear pending state
+        delete s.withdrawals[msg.sender];
+
+        // Transfer underlying SSV back to user
+        if (!SSVStorage.load().token.transfer(msg.sender, amount)) {
+            revert TokenTransferFailed();
+        }
+
+        emit UnstakedWithdrawn(msg.sender, amount);
+    }
+
+    function claimEthRewards() external {
+        StorageStaking storage s = SSVStorageStaking.load();
+        // Update state to calculate latest rewards
+        _syncFees(s);
+        _settle(msg.sender, s);
+
+        uint256 claimable = s.accrued[msg.sender];
+        if (claimable == 0) revert NothingToClaim();
+
+        // Round down to precision supported by protocol storage
```

Comment:
Users with accrued rewards below `DEDUCTED_DIGITS` (10,000,000 wei) cannot claim their rewards at all:

```solidity
uint256 payout = claimable - (claimable % DEDUCTED_DIGITS);
if (payout == 0) revert NothingToClaim();  // Reverts if claimable < DEDUCTED_DIGITS
```

If a user has 9,999,999 wei in rewards, `payout = 0` and the function reverts. This dust is stuck forever if the user unstakes all cSSV.

Suggestion: Allow users to claim any non-zero amount, even if below precision threshold:

```solidity
if (claimable == 0) revert NothingToClaim();
uint256 payout = claimable;  // Claim full amount
s.accrued[msg.sender] = 0;
// Transfer payout
```